### PR TITLE
Accumulo 4374: Modified Shell to fix ChangeSecret tool

### PR DESCRIPTION
**Pull request by milleruntime**

### CODE REVIEW

File: shell/src/main/java/org/apache/accumulo/shell/Shell.java [hunk 1]
```diff
@@ -242,9 +242,8 @@
     }
   }
 
-  public Shell() throws IOException {
-    this(new ConsoleReader());
-  }
+  // no arg constructor should do minimal work since its used in Main ServiceLoader
+  public Shell() throws IOException {}
```
Comment (joshelser) [inline shell/src/main/java/org/apache/accumulo/shell/Shell.java]: Can remove the `throws IOException` as it won't affect callers at all. They were already expecting the IOException -- if we don't throw that anymore, that's fine.

**Comment by joshelser** [comment]
Changes look fine. Were you able to write a test that showed ChangeSecret was broken? I'm not sure if you can use MiniAccumuloCluster to do this:

1. Start MAC
2. Stop MAC
3. Use MAC to exec the ChangeSecret tool
4. Start MAC
5. Ensure that you can interact with it and that the new instance.secret is set

I'm not sure if you'd run into any hurdles around this.

File: shell/src/main/java/org/apache/accumulo/shell/Shell.java (hunk 1 above)
Comment (milleruntime) [inline shell/src/main/java/org/apache/accumulo/shell/Shell.java]: Fixed in 9b46c75a80

**Comment by milleruntime** [comment]
I made multiple attempts at this with no luck.  The problem I am having is
simulating console input to the readPassword called by JCommander in the
ChangeSecret tool.  I can run the class using the exec in
MiniAccumuloClusterImpl but it will hang waiting for input from the
console. I have tried to grab outputstream from the process and write the
passwords to it. Since readPassword is looking for System.in, I also tried
setting the passwords in an InputStream through System.setIn().

The closest I have come to a successful test is by creating a separate
thread that constantly writes the password to the OutputStream of the
ChangeSecret process. This won't allow me to change the password though
since it will be the same thing for both prompts.  I have no way of knowing
if its asking for the old or new password. I tried looking at the
InputStream for which password prompt to determine the correct password but
readLine will also block waiting for the password input.

The ShellServerIT gets around this problem by not calling readLine and just
executing all the commands on the Shell separately.

Another problem I have is that the ChangeSecret tool requires that
Zookeeper be running and not Accumulo, which there is currently no way of
doing with the MAC.

**Comment by joshelser** [comment]
Ok, thanks for trying, @milleruntime! Let me try to run the test with your changes. If it succeeds, I can merge it in.

**Comment by ctubbsii** [comment]
ACCUMULO-4374

**Pull request closed without merging**